protocol/msgpack/raw_metric.go

```diff
@@ -52,7 +52,7 @@ func NewRawMetric(data []byte, readerBufferSize int) aggregated.RawMetric {
 	reader := bytes.NewReader(data)
 	m := &rawMetric{
 		data: data,
-		it:   newBaseIterator(reader, readerBufferSize),
+		it:   newBaseIterator(reader, readerBufferSize, nil),
```

Comment:
The ctor should probably take a baseIteratorOptions passed along from the caller in aggregatedIterator